source/common/http/async_client_impl.cc

```diff
@@ -94,7 +95,7 @@ void AsyncStreamImpl::encodeTrailers(HeaderMapPtr&& trailers) {
 
 void AsyncStreamImpl::sendHeaders(HeaderMap& headers, bool end_stream) {
   headers.insertEnvoyInternalRequest().value(Headers::get().EnvoyInternalRequestValues.True);
-  headers.insertForwardedFor().value(parent_.config_.local_info_.address());
+  Utility::appendXff(headers, *parent_.config_.local_info_.address());
```

Comment:
Is that intended to add address to XFF instead of rewriting one? iirc, if we have 2 addresses in xff isInternal and other things will not work as expected on the upstream Envoy side, but i might be missing something